query-parser/src/lib.rs

```diff
@@ -8,6 +8,11 @@
 // CONDITIONS OF ANY KIND, either express or implied. See the License for the
 // specific language governing permissions and limitations under the License.
 
+#![allow(unused_imports)]
```

Comment:
Oh, perhaps because that import is only used by test code. Indeed; see my note about the test module.

I'll just land that commit, then we can make this one right.